CMakeLists.txt

```diff
@@ -1,100 +1,81 @@
 cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 project(chip_8_cpp VERSION 1.0.0)
 
-if(NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE Release)
-endif()
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake/modules/")
+file(COPY assets DESTINATION ${CMAKE_BINARY_DIR})
 
-add_definitions(-DASSETS_PATH="${PROJECT_SOURCE_DIR}/assets/")
+set(MAIN_TARGET ${PROJECT_NAME}_interpreter)
+add_executable(${MAIN_TARGET} src/main.cpp ${INTERPRETER_SOURCE_FILES})
 
-find_package(SFML 2.3 COMPONENTS graphics audio window system)
+set_property(TARGET ${MAIN_TARGET} PROPERTY CXX_STANDARD 17)
+set_property(TARGET ${MAIN_TARGET} PROPERTY CXX_STANDARD_REQUIRED TRUE)
 
-list(APPEND INTERPRETER_SOURCE_FILES
-    src/Interpreter.cpp
-    src/io/Speaker.cpp
-    src/io/Keypad.cpp
-    src/io/KeyMap.cpp
-    src/io/display/PixelArray.cpp
-    src/io/display/Renderer.cpp
-    src/details/memory.cpp
-    src/details/audio.cpp
-)
+target_include_directories(${MAIN_TARGET} PRIVATE "${PROJECT_SOURCE_DIR}/include")
 
-set(EXECUTABLE_NAME ${PROJECT_NAME}_interpreter)
+include(compiler-warnings)
+target_add_compiler_warnigs(${MAIN_TARGET})
```

Comment:
Variables are evil according to what I've reading about cmake. 
I removed -Wfatal-errors because we are getting to many warnings. We need to do some work before being able to enable that. Probably we will most likely to just disable some warnings.

Reply:
I've updated the flags. No help adding so many... Wall Wextra Wpedantic Wfatal-errors sufficient. What I get really interesting output is from clang-tidy. You need at least clang-8 installed. You can then configure the project with configure_chip8_clang if you source the init script